tests/src/data/use-cases/db-validate-verification-token.spec.ts

```diff
@@ -0,0 +1,122 @@
+import {
+  type LoadGuardianByEmailRepository,
+  type HashComparer
+} from '@/data/protocols'
+import {
+  makeFakeGuardianRepository,
+  makeFakeHashService,
+  mockFakeGuardianLoaded
+} from '@/tests/utils'
+import { VerificationTokenError, NotFoundError } from '@/application/errors'
+import { DbValidateVerificationToken } from '@/data/use-cases/db-validate-verification-token'
+
+interface SutTypes {
+  sut: DbValidateVerificationToken
+  guardianRepositoryStub: LoadGuardianByEmailRepository
+  hashServiceStub: HashComparer
+}
+
+const makeSut = (): SutTypes => {
+  const guardianRepositoryStub = makeFakeGuardianRepository()
+  const hashServiceStub = makeFakeHashService()
+  const sut = new DbValidateVerificationToken(
+    {
+      guardianRepository: guardianRepositoryStub,
+      hashService: hashServiceStub
+    }
+  )
+  return {
+    sut,
+    guardianRepositoryStub,
+    hashServiceStub
+  }
+}
+describe('DbCreateAccessToken UseCase', () => {
+  const fakeValidation = {
+    email: 'any_email@email.com',
+    verificationToken: 'valid_token'
+  }
+
+  describe('tests guardianRepository', () => {
+    it('Should return NotFoundError if not found email is provided', async () => {
+      const { sut, guardianRepositoryStub } = makeSut()
+      jest.spyOn(guardianRepositoryStub, 'loadByEmail').mockResolvedValueOnce(undefined)
+
+      const result = await sut.validate(fakeValidation)
+
+      expect(result).toStrictEqual(new NotFoundError('email'))
+    })
+
+    it('Should return ExpiredVerificationTokenError expired token is provided', async () => {
+      const { sut, guardianRepositoryStub } = makeSut()
+      jest.spyOn(guardianRepositoryStub, 'loadByEmail').mockResolvedValueOnce({
+        ...mockFakeGuardianLoaded(),
+        verificationTokenCreatedAt: new Date(2023, 0, 0, 0, 0, 0)
+      })
+
+      const result = await sut.validate(fakeValidation)
+
+      expect(result).toStrictEqual(new VerificationTokenError())
+    })
+
+    it('Should throw if loadByEmail throws', async () => {
+      const { sut, guardianRepositoryStub } = makeSut()
+      jest.spyOn(guardianRepositoryStub, 'loadByEmail').mockRejectedValueOnce(new Error())
+
+      const promise = sut.validate(fakeValidation)
+
+      await expect(promise).rejects.toThrow()
+    })
+
+    it('Should call loadByEmail with correct email', async () => {
+      const { sut, guardianRepositoryStub } = makeSut()
+      const loadSpy = jest.spyOn(guardianRepositoryStub, 'loadByEmail')
+
+      await sut.validate(fakeValidation)
+
+      expect(loadSpy).toHaveBeenCalledWith(fakeValidation.email)
+    })
+  })
+
+  describe('test HashComparer', () => {
+    it('should return VerificationTokenError if invalid code is provided', async () => {
```

Comment:
could standardize the "should" and "Should"

Reply:
Ok, I will use "Should", because is most used in our project files.